### PR TITLE
Arrow label feature

### PR DESCRIPTION
**Pull request by jrfferreira**
Based on a use case that I have using react-archer, I submit a feature to allow user type text, add symbols or anything else over the arrows. Would be nice to have your consideration over it.

### CODE REVIEW

**Comment by pierpo** [comment]
Nice! Thanks for your contribution. I'll check it out ASAP.

Can you provide a screenshot? 😊

**Comment by jrfferreira** [comment]
Hi @pierpo ,

image

(This is the same screenshot updated in the PR)

**Comment by pierpo** [comment]
@jrfferreira Awesome, thanks 😊

File: src/ArcherElement.js [hunk 1]
```diff
@@ -3,10 +3,13 @@ import PropTypes from 'prop-types';
...
-    if (
-      JSON.stringify(this.props.relations) ===
-      JSON.stringify(nextProps.relations)
-    ) {
+    const currentRelations = (this.props.relations || []).map(r =>
+      (r.label || {}).toString(),
+    );
+    const nextRelations = (nextProps.relations || []).map(r =>
+      (r.label || {}).toString(),
+    );
```
Comment (pierpo) [inline src/ArcherElement.js]: We could do a `stringifyRelation` function called twice to make this clearer 😊

Comment (pierpo) [reply src/ArcherElement.js]: @jrfferreira Ouch, it turns out this function was wrong. We lose the 'to' and 'from' in the stringification. Besides, `toString()` always returns `[Object object]` 😉 
I'll fix this (already merged unfortunately)

Comment (jrfferreira) [reply src/ArcherElement.js]: Oh, sorry about that! I saw the fix too, thanks!

**Pull request merged**